python/hail/docs/tutorials/09-matrixtable.ipynb

```diff
@@ -0,0 +1,490 @@
+    }
+   },
+   "source": [
+    "Let's investigate the genotypes and the call rate.  Let's look at the first few genotypes:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "mt.GT.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "All homozygous reference, which is not surprising.  Let's look at the distribution of genotype calls:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "mt.aggregate_entries(hl.agg.counter(mt.GT.n_alt_alleles()))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Let's compute the call rate directly, and then plot the distribution of call rates per vairant."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "subslide"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "mt.aggregate_entries(hl.agg.fraction(hl.is_defined(mt.GT)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Here's a nice trick: you can use an aggregator inside `annotate_rows` and it will aggregate over columns, that is, summarize the values in the row using the aggregator.  Let's compute and plot call rate per variant."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "mt2 = mt.annotate_rows(call_rate = hl.agg.fraction(hl.is_defined(mt.GT)))\n",
+    "mt2.describe()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "hist_start = 0\n",
+    "hist_end = 1.0\n",
+    "n_bins = 100\n",
+    "bin_size = (hist_end - hist_start) / n_bins\n",
+    "\n",
+    "hist = mt2.aggregate_rows(hl.agg.hist(mt2.call_rate, hist_start, hist_end, n_bins))\n",
+    "\n",
+    "plt.xlim(0.0, 1.0)\n",
+    "plt.bar(hist.bin_edges[:-1], hist.bin_freq, width=bin_size, label='call_rate')\n",
+    "plt.xlabel('Call Rate')\n",
+    "plt.ylabel('Count')\n",
+    "plt.title('Variant Call Rate Histogram')\n",
+    "plt.legend(loc=2)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Exercise: GQ vs DP\n",
+    "\n",
+    "This exercise requires using Hail to learn about a strange property of sequencing datasets.\n",
```

Comment:
In this exercise, you'll use Hail to investigate a strange property of sequencing datasets.